### PR TITLE
chore: update TestDeployment

### PR DESCRIPTION
**Pull request by zirain**
update TestDeployment, so the `default` test output became same as the real world

### CODE REVIEW

File: internal/infrastructure/kubernetes/proxy/resource_provider_test.go [hunk 1]
```diff
@@ -550,6 +544,15 @@ func TestDeployment(t *testing.T) {
 				})
 			}
 
+			if os.Getenv("UPDATE_TESTDATA") != "" {
```
Comment (arkodg) [inline internal/infrastructure/kubernetes/proxy/resource_provider_test.go:547]: can we use this approach https://github.com/envoyproxy/gateway/blob/26d9d9794f0a01bc584bd0273fcfd902467115a0/internal/cmd/egctl/translate_test.go#L29

https://github.com/envoyproxy/gateway/blob/26d9d9794f0a01bc584bd0273fcfd902467115a0/tools/make/golang.mk#L49

**Review by Xunzhuo — COMMENTED** [review]
Have you tested if these files will be updated, each time we ran `make testdata`

**Comment by zirain** [comment]
> Have you tested if these files will be updated, each time we ran `make testdata`

yes

**Comment by Xunzhuo** [comment]
Thanks for confirming, I met this issue when I firstly add `make testdata` to eg.

**Review by arkodg — APPROVED** [review]

**Review by zhaohuabing — APPROVED** [review]

**Pull request merged**